src/generic_core/metrics.ts

```diff
@@ -11,13 +10,19 @@ export interface MetricsData {
   version :number;
   success :number;  // Number of successes for getting access only.
   failure :number;  // Number of failures for getting access only.
+  natType :string;
+  pmpSupport :string;  // Either 'Supported' or 'Not supported'
+  pcpSupport :string;
+  upnpSupport :string;
```

Comment:
This is kinda optional, but I think it would be fine to leave these new fields out of MetricsData.  MetricsData is really just what we read/write to storage.  In the case of success and failure counts this is needed because we want to keep track of the total counts in about one day, even across restarts of uProxy.  But for these new fields, we just query the natType, etc at the time we generate the report, and they never actually need to be saved.